### PR TITLE
Task API gRPC SSL

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Requires #4929

### CODE REVIEW

File: golem/apps/ssl.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        cert_file_name=api.ssl.SERVER_CERT_FILE_NAME,
+        label="App",
+    )
+    shutil.copy(
+        SSLContextConfig.key_and_cert_directory / api.ssl.CLIENT_CERT_FILE_NAME,
+        shared_directory
+    )
+
+
+def create_ssl_context_files(
```
Comment (maaktweluit) [inline golem/apps/ssl.py:62]: this function is not used outside this file, could be private

File: golem/apps/ssl.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        cert = generate_x509_certificate(
+            key,
+            names=['localhost'],
+            ip_addresses=[IPv4Address('127.0.0.1')])
+
+        directory.mkdir(parents=True, exist_ok=True)
+        save_rsa_private_key(key_file, key)
+        save_x509_certificate(cert_file, cert)
+    except Exception:  # pylint: disable=broad-except
+        logger.exception('Unable to create Task API SSL context filex')
```
Comment (maaktweluit) [inline golem/apps/ssl.py:88]: should we log some extra info here like path and of stacktrace?

File: tests/golem/apps/test_ssl.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    create_ssl_context_files,
+    create_golem_ssl_context_files,
+    generate_rsa_private_key,
+    SSLContextConfig,
+)
+from golem.testutils import TempDirFixture
+
+
+@mock.patch.object(SSLContextConfig, 'key_and_cert_directory', None)
+class TestSslContext(TempDirFixture):
```
Comment (maaktweluit) [inline tests/golem/apps/test_ssl.py:18]: i think its better to use the pytest fixture tmpdir ( and tmp_path after we updated pytest )

example: https://github.com/golemfactory/golem/blob/develop/tests/golem/task/test_requestedtaskmanager.py#L51

File: tests/golem/task/test_requestedtaskmanager.py [hunk 4]
```diff
@@ -74,8 +76,8 @@ def setup_method(self, tmpdir, monkeypatch):
             lambda *_: TaskState())
 
     @pytest.mark.asyncio
-    @pytest.mark.freeze_time("1000")
-    async def test_restore_tasks_timedout(self, freezer, mock_client):
+    @pytest.mark.freeze_time("2020")
```
Comment (maaktweluit) [inline tests/golem/task/test_requestedtaskmanager.py:79]: just curious, why did the dates changed? i liked testing the middle ages ;)

**Review by maaktweluit — APPROVED** [review]
LGTM!

Some small comments, still approving

File: golem/apps/ssl.py (hunk 2 above)
Comment (Wiezzel) [reply golem/apps/ssl.py:88]: `logger.exception()` automatically logs stacktrace

File: tests/golem/task/test_requestedtaskmanager.py (hunk 4 above)
Comment (mfranciszkiewicz) [reply tests/golem/task/test_requestedtaskmanager.py:79]: Certificate validation failed due to invalid dates

File: tests/golem/apps/test_ssl.py (hunk 3 above)
Comment (mfranciszkiewicz) [reply tests/golem/apps/test_ssl.py:18]: Simply, these tests are unittest-based, not pytest-based.

File: golem/apps/ssl.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    generate_x509_certificate,
+    save_rsa_private_key,
+    save_x509_certificate,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SSLContextConfig:
+    key_and_cert_directory: Optional[Path] = None
```
Comment (Wiezzel) [inline golem/apps/ssl.py:21]: I'm not really convinced of this global variable approach. Could the path be passed as argument for `TaskComputer`, `AppBenchmarkManager` and `RequestedTaskManager`? This argument could be optional.

**Review by Wiezzel — COMMENTED** [review]
Sorry for late review. The PR looks OK, but I have some doubts about the singleton design.

File: golem/apps/ssl.py (hunk 5 above)
Comment (mfranciszkiewicz) [reply golem/apps/ssl.py:21]: I also had doubts about this approach and personally do not like it. But when I started refactoring, the solution seemed messy and forced. We really need a better way to store runtime configuration.

**Pull request closed without merging**